<a href="https://colab.research.google.com/github/Mariehf64/fev23cda_french_industry/blob/GPS/EDA_GPS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# importation des librairies
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# **EDA du dataset name_geographic_information**

In [ ]:
# importation des données 
source=pd.read_csv('/content/name_geographic_information.csv')

# création d'un copie 
df=source.copy()

In [ ]:
# premières analyses
display(df.head())
display( df.info())

EU_circo  code_région   nom_région chef.lieu_région numéro_département  \
0  Sud-Est           82  Rhône-Alpes             Lyon                 01   
1  Sud-Est           82  Rhône-Alpes             Lyon                 01   
2  Sud-Est           82  Rhône-Alpes             Lyon                 01   
3  Sud-Est           82  Rhône-Alpes             Lyon                 01   
4  Sud-Est           82  Rhône-Alpes             Lyon                 01   

  nom_département       préfecture  numéro_circonscription  \
0             Ain  Bourg-en-Bresse                       1   
1             Ain  Bourg-en-Bresse                       1   
2             Ain  Bourg-en-Bresse                       1   
3             Ain  Bourg-en-Bresse                       1   
4             Ain  Bourg-en-Bresse                       1   

            nom_commune codes_postaux  code_insee   latitude longitude  \
0              Attignat         01340        1024  46.283333  5.166667   
1              Beaupont         01270        1029  46.400000  5.266667   
2                  Bény         01370        1038  46.333333  5.283333   
3             Béreyziat         01340        1040  46.366667      5.05   
4  Bohas-Meyriat-Rignat         01250        1245  46.133333       5.4   

   éloignement  
0         1.21  
1         1.91  
2         1.51  
3         1.71  
4         1.01

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36840 entries, 0 to 36839
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   EU_circo                36840 non-null  object 
 1   code_région             36840 non-null  int64  
 2   nom_région              36840 non-null  object 
 3   chef.lieu_région        36840 non-null  object 
 4   numéro_département      36840 non-null  object 
 5   nom_département         36840 non-null  object 
 6   préfecture              36840 non-null  object 
 7   numéro_circonscription  36840 non-null  int64  
 8   nom_commune             36840 non-null  object 
 9   codes_postaux           36840 non-null  object 
 10  code_insee              36840 non-null  int64  
 11  latitude                33911 non-null  float64
 12  longitude               33999 non-null  object 
 13  éloignement             33878 non-null  float64
dtypes: float64(2), int64(3), object(9)
mem

None

**ANALYSE DES DOUBLONS**



In [ ]:
df.duplicated().sum()

0

**ANALYSE DES NaN**

In [ ]:
rap_na=pd.DataFrame()
rap_na['Nbre_Na']=df.isna().sum()
rap_na['% NAN']=((df.isna().sum()/df.shape[0])*100).round(2)

display(rap_na)

Nbre_Na  % NAN
EU_circo                      0   0.00
code_région                   0   0.00
nom_région                    0   0.00
chef.lieu_région              0   0.00
numéro_département            0   0.00
nom_département               0   0.00
préfecture                    0   0.00
numéro_circonscription        0   0.00
nom_commune                   0   0.00
codes_postaux                 0   0.00
code_insee                    0   0.00
latitude                   2929   7.95
longitude                  2841   7.71
éloignement                2962   8.04

In [ ]:
# nombre d'enregistrement un NaN
print("nombre d'enregistrement contenant au moins un NaN : ",df.isna().any(axis=1).sum())

# enregistrement contenant au moins un NaN
display(df[df.isna().any(axis=1)==1])

nombre d'enregistrement contenant au moins un NaN :  2962


EU_circo  code_région   nom_région chef.lieu_région  \
60       Sud-Est           82  Rhône-Alpes             Lyon   
65       Sud-Est           82  Rhône-Alpes             Lyon   
72       Sud-Est           82  Rhône-Alpes             Lyon   
96       Sud-Est           82  Rhône-Alpes             Lyon   
107      Sud-Est           82  Rhône-Alpes             Lyon   
...          ...          ...          ...              ...   
36835  Outre-Mer            5      Mayotte        Mamoudzou   
36836  Outre-Mer            5      Mayotte        Mamoudzou   
36837  Outre-Mer            5      Mayotte        Mamoudzou   
36838  Outre-Mer            5      Mayotte        Mamoudzou   
36839  Outre-Mer            5      Mayotte        Mamoudzou   

      numéro_département nom_département       préfecture  \
60                    01             Ain  Bourg-en-Bresse   
65                    01             Ain  Bourg-en-Bresse   
72                    01             Ain  Bourg-en-Bresse   
96                    01             Ain  Bourg-en-Bresse   
107                   01             Ain  Bourg-en-Bresse   
...                  ...             ...              ...   
36835                976         Mayotte        Mamoudzou   
36836                976         Mayotte        Mamoudzou   
36837                976         Mayotte        Mamoudzou   
36838                976         Mayotte        Mamoudzou   
36839                976         Mayotte        Mamoudzou   

       numéro_circonscription       nom_commune codes_postaux  code_insee  \
60                          1        Saint-Just         01250        1369   
65                          1     Saint-Sulpice         01340        1387   
72                          1  Treffort-Cuisiat         01370        1426   
96                          2             Brion         01460        1063   
107                         2          Echallon         01130        1152   
...                       ...               ...           ...         ...   
36835                       1     M'TSANGAMOUJI         97600       97613   
36836                       1          OUANGANI         97600       97614   
36837                       1          PAMANDZI         97610       97615   
36838                       1              SADA         97640       97616   
36839                       1          TSINGONI         97600       97617   

       latitude longitude  éloignement  
60          NaN       NaN          NaN  
65          NaN       NaN          NaN  
72          NaN       NaN          NaN  
96          NaN       NaN          NaN  
107         NaN       NaN          NaN  
...         ...       ...          ...  
36835       NaN       NaN          NaN  
36836       NaN       NaN          NaN  
36837       NaN       NaN          NaN  
36838       NaN       NaN          NaN  
36839       NaN       NaN          NaN  

[2962 rows x 14 columns]

**Modification à apporter**

*changement de dtype*
*   code_région en str
*   numéro_circonscription en str
*   code_insee en str

*Correction de valeur*

* certaines valeurs de la variable code_insee doivent être corrigées


*Harmonisation de la police*
* nom_commune en capitalize

*Suprression de variables*

* longitude
* latitude
* eloignement


*Création de variable*
*  longitude : a partir de Base officielle des codes postaux et des communes
* latitude : à partir de Base officielle des codes postaux et des communes


*Changer le nom d'une variable*
* nom de la variable code_insee changé en COM





In [ ]:
# changement dtype

change_dtype_str={'code_région':'str',
              'numéro_circonscription':'str',
              'code_insee':'str',
              }
df=df.astype(change_dtype_str)

# correction de valeur

df.code_insee=df.code_insee.apply(lambda row : row.split('.')[0])
df.code_insee=df.code_insee.apply(lambda row : row.zfill(5))


#harmonisation de la police

df.nom_commune=df.nom_commune.apply(lambda row : row.capitalize())

# suppression de variables

df=df.iloc[:,:11]

# changer le nom d'une variable

df=df.rename({'code_insee':'COM'},axis=1)

# vérification 

display(df.head())
df.info()

EU_circo code_région   nom_région chef.lieu_région numéro_département  \
0  Sud-Est          82  Rhône-Alpes             Lyon                 01   
1  Sud-Est          82  Rhône-Alpes             Lyon                 01   
2  Sud-Est          82  Rhône-Alpes             Lyon                 01   
3  Sud-Est          82  Rhône-Alpes             Lyon                 01   
4  Sud-Est          82  Rhône-Alpes             Lyon                 01   

  nom_département       préfecture numéro_circonscription  \
0             Ain  Bourg-en-Bresse                      1   
1             Ain  Bourg-en-Bresse                      1   
2             Ain  Bourg-en-Bresse                      1   
3             Ain  Bourg-en-Bresse                      1   
4             Ain  Bourg-en-Bresse                      1   

            nom_commune codes_postaux    COM  
0              Attignat         01340  01024  
1              Beaupont         01270  01029  
2                  Bény         01370  01038  
3             Béreyziat         01340  01040  
4  Bohas-meyriat-rignat         01250  01245

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36840 entries, 0 to 36839
Data columns (total 11 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   EU_circo                36840 non-null  object
 1   code_région             36840 non-null  object
 2   nom_région              36840 non-null  object
 3   chef.lieu_région        36840 non-null  object
 4   numéro_département      36840 non-null  object
 5   nom_département         36840 non-null  object
 6   préfecture              36840 non-null  object
 7   numéro_circonscription  36840 non-null  object
 8   nom_commune             36840 non-null  object
 9   codes_postaux           36840 non-null  object
 10  COM                     36840 non-null  object
dtypes: object(11)
memory usage: 3.1+ MB


**coordonnées GPS**


source officielle reactulisée : Base officielle des codes postaux et des communes (BOCP)

page de référence : https://datanova.laposte.fr/explore/dataset/laposte_hexasmal/information/?disjunctive.code_commune_insee&disjunctive.nom_de_la_commune&disjunctive.code_postal&disjunctive.ligne_5

dataset de référence : https://datanova.laposte.fr/api/explore/v2.1/catalog/datasets/laposte_hexasmal/exports/csv?lang=fr&timezone=Europe%2FBerlin&use_labels=true&delimiter=%3B

**Variables :**

* Code_commune_INSEE: code_commune_insee[text]<br>
* Nom_commune: nom_de_la_commune[text]<br>
* Code_postal: code_postal[text]<br>
* Ligne_5: ligne_5[text] Ligne 5 dans l’écriture de l’adresse, utilisée  notamment pour préciser l’ancienne commune ou le lieu-dit<br>
* Libellé_d_acheminement: libelle_d_acheminement[text]<br>
* coordonnees_geographiques: coordonnees_geographiques[text]<br>


In [ ]:
df_gps=pd.read_csv('https://datanova.laposte.fr/api/explore/v2.1/catalog/datasets/laposte_hexasmal/exports/csv?lang=fr&timezone=Europe%2FBerlin&use_labels=true&delimiter=%3B',sep=';')

**Premières analyses**

In [ ]:
print(df_gps.info())
df_gps=df_gps.sort_values(by='Code_commune_INSEE')
display(df_gps.head())

<class 'pandas.core.frame.DataFrame'>
Int64Index: 39193 entries, 4039 to 15707
Data columns (total 6 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   Code_commune_INSEE         39193 non-null  object
 1   Nom_commune                39193 non-null  object
 2   Code_postal                39193 non-null  int64 
 3   Ligne_5                    4696 non-null   object
 4   Libellé_d_acheminement     39193 non-null  object
 5   coordonnees_geographiques  39193 non-null  object
dtypes: int64(1), object(5)
memory usage: 2.1+ MB
None


Code_commune_INSEE              Nom_commune  Code_postal Ligne_5  \
4039               01001  L ABERGEMENT CLEMENCIAT         1400     NaN   
31363              01002    L ABERGEMENT DE VAREY         1640     NaN   
17384              01004        AMBERIEU EN BUGEY         1500     NaN   
31364              01005      AMBERIEUX EN DOMBES         1330     NaN   
31365              01006                  AMBLEON         1300     NaN   

        Libellé_d_acheminement  coordonnees_geographiques  
4039   L ABERGEMENT CLEMENCIAT  46.153721024, 4.925850148  
31363    L ABERGEMENT DE VAREY  46.009605679, 5.428087796  
17384        AMBERIEU EN BUGEY  45.961048852, 5.372275427  
31364      AMBERIEUX EN DOMBES   45.99616357, 4.911967138  
31365                  AMBLEON  45.749886304, 5.594584599

In [ ]:
print("nombre de modalité de la variable code_commune_insee :",len(df_gps.Code_commune_INSEE.unique()))
print("nombre de modalité de la variable coordonnees_geographiques :",len(df_gps.coordonnees_geographiques.unique()))


nombre de modalité de la variable code_commune_insee : 35079
nombre de modalité de la variable coordonnees_geographiques : 35079


**Modifier le set df_gps**

*Variables conservées :*
* Code_commune_INSEE:
* Code_postal

*variables créées:* 

* latitude : a partir de coordonnees_geographiques
* longitude : a partir de  coordonnees_geographiques

*Nom de variable à modifier*

* Code_commune_INSEE en COM

*Dtype à modifier*

* dtype de Code_postal modifié en str

*Valeur à modifier*

* certaines valeurs de la variable Code_postal doivent être corrigées



In [ ]:
df_gps['latitude']=df_gps.coordonnees_geographiques.apply(lambda row : row.split(',')[0])
df_gps['longitude']=df_gps.coordonnees_geographiques.apply(lambda row : row.split(' ')[1])

In [ ]:
df_gps=df_gps[['Code_commune_INSEE','Code_postal','latitude','longitude']]
df_gps=df_gps.rename({'Code_commune_INSEE':'COM'},axis=1)
df_gps.Code_postal=df_gps.Code_postal.astype(str)

In [ ]:
print(df_gps.info())
display(df_gps.head())

<class 'pandas.core.frame.DataFrame'>
Int64Index: 39193 entries, 4039 to 15707
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   COM          39193 non-null  object
 1   Code_postal  39193 non-null  object
 2   latitude     39193 non-null  object
 3   longitude    39193 non-null  object
dtypes: object(4)
memory usage: 1.5+ MB
None


COM Code_postal      latitude    longitude
4039   01001        1400  46.153721024  4.925850148
31363  01002        1640  46.009605679  5.428087796
17384  01004        1500  45.961048852  5.372275427
31364  01005        1330   45.99616357  4.911967138
31365  01006        1300  45.749886304  5.594584599

In [ ]:
df_test=df.copy()

df_test=df_test.merge(right=df_gps, on='COM', how='left')

print(df_test.info())
print(df_test.isna().sum())


<class 'pandas.core.frame.DataFrame'>
Int64Index: 40890 entries, 0 to 40889
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   EU_circo                40890 non-null  object
 1   code_région             40890 non-null  object
 2   nom_région              40890 non-null  object
 3   chef.lieu_région        40890 non-null  object
 4   numéro_département      40890 non-null  object
 5   nom_département         40890 non-null  object
 6   préfecture              40890 non-null  object
 7   numéro_circonscription  40890 non-null  object
 8   nom_commune             40890 non-null  object
 9   codes_postaux           40890 non-null  object
 10  COM                     40890 non-null  object
 11  Code_postal             38730 non-null  object
 12  latitude                38730 non-null  object
 13  longitude               38730 non-null  object
dtypes: object(14)
memory usage: 4.7+ MB
None
EU_circo     

In [ ]:
df_test['GPS']=df_test.latitude+','+df_test.longitude
print("nombre de coordonnées uniques : ", len(df_test.GPS.unique()))


nombre de coordonnées uniques :  34564


# **Création d'un dataset code_geo à partir de sources primaires**

**PREMIERE ETAPE : CREATION DU NIVEAU COLLECTIVITEE** 

Page web de référence : https://www.insee.fr/fr/information/6051727

Source de référence : https://www.insee.fr/fr/statistiques/fichier/6051727/commune_2022.csv

Liste des communes, arrondissements municipaux, communes déléguées et communes associées au 1er janvier 2022, avec le code des niveaux supérieurs (canton ou pseudo-canton, département, région)

Nom de la variable	Longueur de la variable	Désignation et modalités de la variable<br>
TYPECOM	 :  Type de commune <br>
COM	 :	Code commune<br>
REG	 :	Code région <br>
DEP	 :	Code département <br>
CTCD	:	Code de la collectivité territoriale ayant les compétences  départementales <br>
ARR	:	Code arrondissement <br>
TNCC :	Type de nom en clair <br>
NCC	:	Nom en clair (majuscules) <br>
NCCENR	: 	Nom en clair (typographie riche) <br>
LIBELLE	:	Nom en clair (typographie riche) avec article <br>
CAN	:	Code canton. Pour les communes « multi-cantonales » code décliné de 99 à 90 (pseudo-canton) ou de 89 à 80 (communes nouvelles) <br>
COMPARENT	:	Code de la commune parente pour les arrondissements municipaux et les communes associées ou déléguées. <br>

In [ ]:
df_com=pd.read_csv('https://www.insee.fr/fr/statistiques/fichier/6051727/commune_2022.csv')

**Premières analyses**

In [ ]:
display(df_com.info())
display(df_com.head())
display(df_com[df_com.REG.isna()==1])

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 37601 entries, 0 to 37600
Data columns (total 12 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   TYPECOM    37601 non-null  object 
 1   COM        37601 non-null  object 
 2   REG        35000 non-null  float64
 3   DEP        35000 non-null  object 
 4   CTCD       35000 non-null  object 
 5   ARR        34983 non-null  object 
 6   TNCC       37601 non-null  int64  
 7   NCC        37601 non-null  object 
 8   NCCENR     37601 non-null  object 
 9   LIBELLE    37601 non-null  object 
 10  CAN        34855 non-null  object 
 11  COMPARENT  2646 non-null   float64
dtypes: float64(2), int64(1), object(9)
memory usage: 3.4+ MB


None

TYPECOM    COM   REG DEP CTCD  ARR  TNCC                    NCC  \
0     COM  01001  84.0  01  01D  012     5  ABERGEMENT CLEMENCIAT   
1     COM  01002  84.0  01  01D  011     5    ABERGEMENT DE VAREY   
2     COM  01004  84.0  01  01D  011     1      AMBERIEU EN BUGEY   
3     COM  01005  84.0  01  01D  012     1    AMBERIEUX EN DOMBES   
4     COM  01006  84.0  01  01D  011     1                AMBLEON   

                  NCCENR                  LIBELLE   CAN  COMPARENT  
0  Abergement-Clémenciat  L'Abergement-Clémenciat  0108        NaN  
1    Abergement-de-Varey    L'Abergement-de-Varey  0101        NaN  
2      Ambérieu-en-Bugey        Ambérieu-en-Bugey  0101        NaN  
3    Ambérieux-en-Dombes      Ambérieux-en-Dombes  0122        NaN  
4                Ambléon                  Ambléon  0104        NaN

TYPECOM    COM  REG  DEP CTCD  ARR  TNCC                       NCC  \
14       COMD  01015  NaN  NaN  NaN  NaN     1                 ARBIGNIEU   
23       COMD  01025  NaN  NaN  NaN  NaN     0             BAGE LA VILLE   
32       COMD  01033  NaN  NaN  NaN  NaN     0  BELLEGARDE SUR VALSERINE   
36       COMD  01036  NaN  NaN  NaN  NaN     0         BELMONT LUTHEZIEU   
57       COMD  01059  NaN  NaN  NaN  NaN     0                    BRENAZ   
...       ...    ...  ...  ...  ...  ...   ...                       ...   
36854    COMD  89473  NaN  NaN  NaN  NaN     0       VILLIERS SUR THOLON   
36857    COMA  89476  NaN  NaN  NaN  NaN     3                  VILLOTTE   
36865    COMD  89484  NaN  NaN  NaN  NaN     0                    VOLGRE   
37034    COMD  91222  NaN  NaN  NaN  NaN     1                 ESTOUCHES   
37077    COMD  91390  NaN  NaN  NaN  NaN     0                 MEREVILLE   

                         NCCENR                   LIBELLE  CAN  COMPARENT  
14                    Arbignieu                 Arbignieu  NaN     1015.0  
23                Bâgé-la-Ville             Bâgé-la-Ville  NaN     1025.0  
32     Bellegarde-sur-Valserine  Bellegarde-sur-Valserine  NaN     1033.0  
36            Belmont-Luthézieu         Belmont-Luthézieu  NaN     1036.0  
57                       Brénaz                    Brénaz  NaN     1453.0  
...                         ...                       ...  ...        ...  
36854       Villiers-sur-Tholon       Villiers-sur-Tholon  NaN    89003.0  
36857                  Villotte               La Villotte  NaN    89472.0  
36865                    Volgré                    Volgré  NaN    89003.0  
37034                 Estouches                 Estouches  NaN    91390.0  
37077                 Méréville                 Méréville  NaN    91390.0  

[2601 rows x 12 columns]



*  La base commune de l'Insee contient au 1er janvier 2022 : 37600 entrées
*  Pour rappel, la base name_geographique_information contient 36840 entrées
* 35 000 enregistrement sont renseignés au niveau REG et DEP




**Transformation de la base df_com**

***variables conservées***

TYPECOM	 :  Type de commune <br>
COM	 :	Code commune<br>
REG	 :	Code région <br>
DEP	 :	Code département <br>
LIBELLE	:	Nom en clair (typographie riche) avec article <br>



***variables supprimées*** 

CTCD	:	Code de la collectivité ayant les compétences  départementales <br>
TNCC :	Type de nom en clair <br>
NCC	:	Nom en clair (majuscules) <br>
NCCENR	: 	Nom en clair (typographie riche) <br>
CAN	:	Code canton  <br>
COMPARENT	:	Code de la commune parente <br>
ARR	:	Code arrondissement <br>


***Changement de Dtype***

la dtype de la variable REG est changer en objet


***Changement nom de la variable***

la nom de la variable LIBELLE est changé en LIBELLE_COM




In [ ]:
df_com['REG']=df_com['REG'].astype(str) # changement de la dtype de REG

df_com.REG=df_com.REG.astype(str).apply(lambda x : x.split('.')[0]) # suppression de la décimale

df_com=df_com[['TYPECOM','COM','REG','DEP','LIBELLE']] # suppression/conservation de variables

df_com=df_com.rename({'LIBELLE':'LIBELLE_COM'},axis=1) # changement du nom de la variable LIBELLE_COM


In [ ]:
df_com.REG=df_com.REG.replace({'nan':np.nan})  # modification des 'nan' en NaN

In [ ]:
print(df_com.info()) # vérification

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 37601 entries, 0 to 37600
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   TYPECOM      37601 non-null  object
 1   COM          37601 non-null  object
 2   REG          35000 non-null  object
 3   DEP          35000 non-null  object
 4   LIBELLE_COM  37601 non-null  object
dtypes: object(5)
memory usage: 1.4+ MB
None


**DEUXIEME ETAPE : CREATION DU NIVEAU DEPARTEMENT**

Page web de référence : https://www.insee.fr/fr/information/6051727

Source de référence : https://www.insee.fr/fr/statistiques/fichier/6051727/departement_2022.csv

***Variables :***

DEP	:	Code département <br>
REG	:	Code région <br>
CHEFLIEU :	Code de la commune chef-lieu <br>
TNCC	:	Type de nom en clair <br>
NCC	:	Nom en clair (majuscules) <br>
NCCENR	:	Nom en clair (typographie riche) <br>
LIBELLE	:	Nom en clair (typographie riche) avec article
Régions <br>


In [ ]:
df_dep=pd.read_csv('https://www.insee.fr/fr/statistiques/fichier/6051727/departement_2022.csv')

**Premières analyses**

In [ ]:
print(df_dep.info())
print(df_dep.head())


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 101 entries, 0 to 100
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   DEP       101 non-null    object
 1   REG       101 non-null    int64 
 2   CHEFLIEU  101 non-null    object
 3   TNCC      101 non-null    int64 
 4   NCC       101 non-null    object
 5   NCCENR    101 non-null    object
 6   LIBELLE   101 non-null    object
dtypes: int64(2), object(5)
memory usage: 5.6+ KB
None
  DEP  REG CHEFLIEU  TNCC                      NCC                   NCCENR  \
0  01   84    01053     5                      AIN                      Ain   
1  02   32    02408     5                    AISNE                    Aisne   
2  03   84    03190     5                   ALLIER                   Allier   
3  04   93    04070     4  ALPES DE HAUTE PROVENCE  Alpes-de-Haute-Provence   
4  05   93    05061     4             HAUTES ALPES             Hautes-Alpes   

                   LIBEL

*   La base insee departement contient au 1er janvier 2022 101 entrées

**Transformation de la df_dep**

***Variables conservé :***

DEP : Code département <br>
LIBELLE : Nom en clair (typographie riche) avec article Régions<br>

**Variables supprimées**<br>
REG : Code région<br>
CHEFLIEU : Code de la commune chef-lieu<br>
TNCC : Type de nom en clair<br>
NCC : Nom en clair (majuscules)<br>
NCCENR : Nom en clair (typographie riche)<br>

**Nom de variable changé**<br>

le nom de la variable LIBELLE est changé en LIBELLE_DEP




In [ ]:
df_dep=df_dep[['DEP','LIBELLE']] # selection/suppression de variables
df_dep=df_dep.rename({'LIBELLE':'LIBELLE_DEP'},axis=1) # modification du nom de la variable LIBELLE en LIBELLE_DEP
df_dep.info() # vérification

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 101 entries, 0 to 100
Data columns (total 2 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   DEP          101 non-null    object
 1   LIBELLE_DEP  101 non-null    object
dtypes: object(2)
memory usage: 1.7+ KB


**TROISIEME ETAPE : CREATION DU NIVEAU REGION**

Page web de référence : https://www.insee.fr/fr/information/6051727

Source de référence : https://www.insee.fr/fr/statistiques/fichier/6051727/region_2022.csv

***Variables :*** <br>
REG	:	Code région <br>
CHEFLIEU :	Code de la commune chef-lieu <br>
TNCC	:	Type de nom en clair <br>
NCC	:	Nom en clair (majuscules) <br>
NCCENR	:	Nom en clair (typographie riche) <br>
LIBELLE	:	Nom en clair (typographie riche) avec article
Régions <br>


In [ ]:
df_reg=pd.read_csv('https://www.insee.fr/fr/statistiques/fichier/6051727/region_2022.csv')

**Premières analyses**


In [ ]:
print(df_reg.info())
print(df_reg.head())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18 entries, 0 to 17
Data columns (total 6 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   REG       18 non-null     int64 
 1   CHEFLIEU  18 non-null     object
 2   TNCC      18 non-null     int64 
 3   NCC       18 non-null     object
 4   NCCENR    18 non-null     object
 5   LIBELLE   18 non-null     object
dtypes: int64(2), object(4)
memory usage: 992.0+ bytes
None
   REG CHEFLIEU  TNCC         NCC      NCCENR     LIBELLE
0    1    97105     3  GUADELOUPE  Guadeloupe  Guadeloupe
1    2    97209     3  MARTINIQUE  Martinique  Martinique
2    3    97302     3      GUYANE      Guyane      Guyane
3    4    97411     0  LA REUNION  La Réunion  La Réunion
4    6    97608     0     MAYOTTE     Mayotte     Mayotte


**modifier la base df_reg**

***Variables conservées*** <br>
REG : Code région <br>
LIBELLE : Nom en clair (typographie riche) avec article Régions<br>

***Dtype modidié*** <br>
le dtype de REG est modifié en objet

***Nom de variable modifié***<br>
le nom de la variable LIBELLE est modifié en LIBELLE_REG

***Variables supprimées***<br>
CHEFLIEU : Code de la commune chef-lieu<br>
TNCC : Type de nom en clair<br>
NCC : Nom en clair (majuscules)<br>
NCCENR : Nom en clair (typographie riche)<br>

In [ ]:
df_reg=df_reg[['REG','LIBELLE']] # selection/suppresion de variables

df_reg.REG=df_reg.REG.astype('str') # dtype de REG modifié en object
df_reg=df_reg.rename({'LIBELLE':'LIBELLE_REG'},axis=1)

df_reg.info() # vérification

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18 entries, 0 to 17
Data columns (total 2 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   REG          18 non-null     object
 1   LIBELLE_REG  18 non-null     object
dtypes: object(2)
memory usage: 416.0+ bytes


<ipython-input-279-14e4f1757674>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_reg.REG=df_reg.REG.astype('str') # dtype de REG modifié en object


**CREATION DE LA BASE CODE GEO**

***Création de la base au niveau communal***

In [ ]:
code_geo=df_com.copy() # création de la base à partir du niveau commune

**Ajout du niveau départemental**

In [ ]:
code_geo=code_geo.merge(right=df_dep,on='DEP',how='left') # ajout du niveau département

**Analyse du merge niveau départemental**

In [ ]:
code_geo.info() # vérification

<class 'pandas.core.frame.DataFrame'>
Int64Index: 37601 entries, 0 to 37600
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   TYPECOM      37601 non-null  object
 1   COM          37601 non-null  object
 2   REG          35000 non-null  object
 3   DEP          35000 non-null  object
 4   LIBELLE_COM  37601 non-null  object
 5   LIBELLE_DEP  35000 non-null  object
dtypes: object(6)
memory usage: 2.0+ MB


In [ ]:
print("nombre d'enregistrements : ",code_geo.shape[0])
print("Nombre d'enregistrements correctement renseignés : ",code_geo[code_geo.isna().any(axis=1)==0].shape[0])
print("Nombre d'enregistrements avec NaN",code_geo.isna().any(axis=1).sum())
print("TYPECOM pour lesquels, il y des NaN : ",code_geo.TYPECOM[code_geo.isna().any(axis=1)==1].unique())

nombre d'enregistrements :  37601
Nombre d'enregistrements correctement renseignés :  35000
Nombre d'enregistrements avec NaN 2601
TYPECOM pour lesquels, il y des NaN :  ['COMD' 'COMA']


Conformément à la variable DEP de div_com ,on obtient 35 000 enregistrements correctement renseignés 


Les enregistrements avec NaN ont pour modalité sur la variable TYPECOM : 
* COMA :Commune associée <br>
* COMD :Commune déléguée <br>

***Ajout du niveau régional***

In [ ]:
code_geo=code_geo.merge(right=df_reg, on='REG',how='left')

***Analyse du merge du niveau régional***

In [ ]:
code_geo.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 37601 entries, 0 to 37600
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   TYPECOM      37601 non-null  object
 1   COM          37601 non-null  object
 2   REG          35000 non-null  object
 3   DEP          35000 non-null  object
 4   LIBELLE_COM  37601 non-null  object
 5   LIBELLE_DEP  35000 non-null  object
 6   LIBELLE_REG  35000 non-null  object
dtypes: object(7)
memory usage: 2.3+ MB


In [ ]:
print("nombre d'enregistrements : ",code_geo.shape[0])
print("Nombre d'enregistrements correctement renseignés : ",code_geo[code_geo.isna().any(axis=1)==0].shape[0])
print("Nombre d'enregistrements avec NaN",code_geo.isna().any(axis=1).sum())
print("TYPECOM pour lesquels, il y des NaN : ",code_geo.TYPECOM[code_geo.isna().any(axis=1)==1].unique())

nombre d'enregistrements :  37601
Nombre d'enregistrements correctement renseignés :  35000
Nombre d'enregistrements avec NaN 2601
TYPECOM pour lesquels, il y des NaN :  ['COMD' 'COMA']


Conformément à la variable REG de div_com ,on obtient 35 000 enregistrements correctement renseignés 


Les enregistrements avec NaN ont pour modalité sur la variable TYPECOM : 
* COMA :Commune associée <br>
* COMD :Commune déléguée <br>

**Ajout des coordonnées GPS**


In [ ]:
geo=code_geo.copy()

geo=geo.merge(right=df_gps, on='COM',how='left')

display(geo.head())
print(geo.info())
print(geo.isna().sum())

TYPECOM    COM REG DEP              LIBELLE_COM LIBELLE_DEP  \
0     COM  01001  84  01  L'Abergement-Clémenciat         Ain   
1     COM  01002  84  01    L'Abergement-de-Varey         Ain   
2     COM  01004  84  01        Ambérieu-en-Bugey         Ain   
3     COM  01005  84  01      Ambérieux-en-Dombes         Ain   
4     COM  01006  84  01                  Ambléon         Ain   

            LIBELLE_REG Code_postal      latitude    longitude  
0  Auvergne-Rhône-Alpes        1400  46.153721024  4.925850148  
1  Auvergne-Rhône-Alpes        1640  46.009605679  5.428087796  
2  Auvergne-Rhône-Alpes        1500  45.961048852  5.372275427  
3  Auvergne-Rhône-Alpes        1330   45.99616357  4.911967138  
4  Auvergne-Rhône-Alpes        1300  45.749886304  5.594584599

<class 'pandas.core.frame.DataFrame'>
Int64Index: 43121 entries, 0 to 43120
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   TYPECOM      43121 non-null  object
 1   COM          43121 non-null  object
 2   REG          38937 non-null  object
 3   DEP          38937 non-null  object
 4   LIBELLE_COM  43121 non-null  object
 5   LIBELLE_DEP  38937 non-null  object
 6   LIBELLE_REG  38937 non-null  object
 7   Code_postal  41084 non-null  object
 8   latitude     41084 non-null  object
 9   longitude    41084 non-null  object
dtypes: object(10)
memory usage: 3.6+ MB
None
TYPECOM           0
COM               0
REG            4184
DEP            4184
LIBELLE_COM       0
LIBELLE_DEP    4184
LIBELLE_REG    4184
Code_postal    2037
latitude       2037
longitude      2037
dtype: int64


**Analyse du merge des coordonnées GPS** 

In [ ]:
print("Nombre d'enregistrements correctement renseignés : ",code_geo[code_geo.isna().any(axis=1)==0].shape[0])
print("Nombre d'enregistrements avec NaN",code_geo.isna().any(axis=1).sum())
print("TYPECOM pour lesquels, il y des NaN : ",code_geo.TYPECOM[code_geo.isna().any(axis=1)==1].unique())

Nombre d'enregistrements correctement renseignés :  38926
Nombre d'enregistrements avec NaN 4195
TYPECOM pour lesquels, il y des NaN :  ['COMD' 'COM' 'COMA']


In [ ]:
code_geo.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 43121 entries, 0 to 43120
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   TYPECOM      43121 non-null  object
 1   COM          43121 non-null  object
 2   REG          43121 non-null  object
 3   DEP          38937 non-null  object
 4   LIBELLE_COM  43121 non-null  object
 5   LIBELLE_DEP  38937 non-null  object
 6   LIBELLE_REG  38937 non-null  object
 7   Code_postal  41084 non-null  object
 8   latitude     41084 non-null  object
 9   longitude    41084 non-null  object
dtypes: object(10)
memory usage: 3.6+ MB
